pkg/controller/windowsmachine/metrics/metrics.go

```diff
@@ -84,19 +85,37 @@ func Add(ctx context.Context, cfg *rest.Config, namespace string) error {
 		return errors.Wrap(err, "could not create metrics Service")
 	}
 
-	// CreateServiceMonitors will automatically create the prometheus-operator ServiceMonitor resources
-	// necessary to configure Prometheus to scrape metrics from this operator.
-	services := []*v1.Service{service}
-	_, err = metrics.CreateServiceMonitors(cfg, namespace, services)
+	// Create a monitoring client
+	mclient, err := monclient.NewForConfig(cfg)
 	if err != nil {
-		log.Error(err, "could not create ServiceMonitor object")
-		// If this operator is deployed to a cluster without the prometheus-operator running, it will return
-		// ErrServiceMonitorNotPresent, which can be used to safely skip ServiceMonitor creation.
-		if err == metrics.ErrServiceMonitorNotPresent {
-			metricsEnabled = false
-			return errors.Wrap(err, "install prometheus-operator in your cluster to create ServiceMonitor objects")
+		return errors.Wrap(err, "could not create monitoring client")
+	}
 
+	// If ServiceMonitor object does not exist, create one
+	// else update the current one.
+	existingSM, err := mclient.ServiceMonitors(namespace).Get(context.TODO(), windowsMetricsEndpoints, metav1.GetOptions{})
+	if existingSM != nil {
+		_ = mclient.ServiceMonitors(namespace).Delete(context.TODO(), windowsMetricsEndpoints, metav1.DeleteOptions{})
+		services := []*v1.Service{service}
+		_, err = metrics.CreateServiceMonitors(cfg, namespace, services)
```

Comment:
should we return an error here ? something like `could not create ServiceMonitor object`